src/explorer/weights/WeightSlider.js

```diff
@@ -2,30 +2,61 @@
 
 import React from "react";
 
-export const WEIGHT_SLIDER_MIN = -5;
-export const WEIGHT_SLIDER_MAX = 5;
+/**
+ * The Weight is the number that the WeightSlider actually provides to the
+ * instantiating component. The Weight varies in the range
+ * [0, 2**MAX_SLIDER_VALUE].
+ */
+export type Weight = number;
```

Comment:
Did you consider making `Weight` and `SliderPosition` opaque subtypes of
`number`? As it stands, in all modules they are interchangeable.

Reply:
I could do that, but IMO it adds a little bit of complexity without any real benefit, because the module where "all of the action happens" (WeightSlider.js) will still allow improper conversions. And exporting the types from a separate module doesn't seem worth the effort. Underlying these tradeoffs: the SliderPosition isn't used elsewhere in the codebase (it is exported for testing purposes only) so if we did invest the effort to make this change robustly, we would not benefit from it.